### DISCUSSION TITLE
Sets inside columns

### DISCUSSION THREAD

**Discussion by jeremyvienney** [jacksleight/statamic-bard-texstyle#36] [General]
Hi,

I was playing with the addon which is great work by the way specially for the div wrapping node.
As I can understand the code about the div node, it create a new node by wrapping in a div with div[data-bts].
I thought the config property ``content: 'block+',`` would let us create anything inside the columns but it seems that we cannot have the "+" button to inject a set. 
Is this something that is missing or functionnal on your side but not mine, or is this something very hard to implement by a limitation of tiptap ?

**Comment by jacksleight** [comment]
Hey. Yeah this is actually a limitation of Bard, it only supports adding sets at the top level of the content. It's the same reason you can't add sets inside things like tables (which also allow `block+`).

**Reply by jeremyvienney** [reply]
Thanks,
Ah ok I thought it was more on the titap side.
You must have more experience than me with Bard and the team, do you know if this is a complex limitation to fix?
It would be so nice if we could select which sets can be included at a level other than root, with a little customization on our part to adapt the preview mode to make editing usable we would have a really nice editor!
I'll try to see on the V4 discord when Jack is online what he thinks.

**Reply by jacksleight** [reply]
As far as Tiptap goes I don't think it would be too tricky (sets are just another node after all). But Bard augments the data into a flat array of HTML blocks and set blocks, which you then loop over in your template. If you had sets buried inside HTML elements I'm not sure how the current templating would work. And the UI/UX could be tricky, it would be very easy to make a mess by inserting sets into places where they don't really work/fit (like a really narrow table or div column).

Out of interest, what kind of things are you looking to add inside other elements? I have plans for a future Texstyle Pro feature that might help with this (not making any promises right now, its a way off 😄).

**Comment by jeremyvienney** [comment]
Indeed it is augmented in html block for {{ text }} but maybe the parser could make an exception if he find a "column" div inside and stop html parsing and render a dedicated set where we could customize the template then.
From the config Bard file, I thought for exemple to a toggle where a set could be inserted elsewhere at the top level, for sets like a custom button, custom picture with styles options like filters, layers bg, overflow position etc.

**Comment by jacksleight** [comment]
Just doing some housekeeping and going to close this one now.

You can now add [pins](https://jacksleight.dev/docs/bard-texstyle/pins) inside divs, which are basically sets but inline.
